### PR TITLE
Add new illuminants

### PR DESCRIPTION
**Pull request by BierretA**
## Description

This small PR adds new white points for color conversions. See Issue #5223 .

I added illuminants B and C, as well as a new observer compatible with color conversions done with the R language.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/color/colorconv.py [hunk 1]
```diff
@@ -415,30 +418,43 @@ def hsv2rgb(hsv):
...
     """Get the XYZ coordinates of the given illuminant and observer [1]_.
 
     Parameters
     ----------
-    illuminant : {"A", "D50", "D55", "D65", "D75", "E"}, optional
+    illuminant : {"A", "B", "C", "D50", "D55", "D65", "D75", "E"}, optional
         The name of the illuminant (the function is NOT case sensitive).
-    observer : {"2", "10"}, optional
+    observer : {"2", "10", "R"}, optional
         The aperture angle of the observer.
```
Comment (mkcor) [inline skimage/color/colorconv.py:458]:
```suggestion
        One of: 2-degree observer, 10-degree observer, or 'R' observer as in
        R function grDevices::convertColor.
```

File: skimage/color/colorconv.py [hunk 2]
```diff
@@ -910,9 +927,9 @@ def xyz2lab(xyz, illuminant="D65", observer="2"):
     ----------
     xyz : (..., 3) array_like
         The image in XYZ format. Final dimension denotes channels.
-    illuminant : {"A", "D50", "D55", "D65", "D75", "E"}, optional
+    illuminant : {"A", "B", "C", "D50", "D55", "D65", "D75", "E"}, optional
         The name of the illuminant (the function is NOT case sensitive).
-    observer : {"2", "10"}, optional
+    observer : {"2", "10", "R"}, optional
         The aperture angle of the observer.
```
Comment (mkcor) [inline skimage/color/colorconv.py:933]: Update here as well, etc.

**Review by mkcor — COMMENTED** [review]
Exciting addition, thanks!

**Comment by BierretA** [comment]
Unit tests work on my system but not with github actions.
It looks likes some tests fail because Pooch cannot fetch the new data files. I do not know why though.

**Comment by mkcor** [comment]
Hello @BierretA,

It definitely looks like an issue with Pooch via GitHub Actions (or vice versa). Tests run fine locally for me as well. Looking into the first failure (`TestColorconv.test_lab2xyz`), we can see that, [over GitHub Actions](https://pipelines.actions.githubusercontent.com/ITGOmoHsVjtvD2vjejJjVxoCD9oCbGYrPOYQanJNnGN0KqRbGi/_apis/pipelines/1/runs/763/signedlogcontent/4?urlExpires=2021-02-25T09%3A56%3A49.5975706Z&urlSigningMethod=HMACV1&urlSignature=BVvEDMZ6%2ByGXMLvcC5ISat9URrSUiMZXlOp26Wm3bvg%3D), Pooch is attempting the following
```
Downloading file 'color/tests/data/lab_array_a_10.npy' from 'https://github.com/scikit-image/scikit-image/raw/main/skimage/color/tests/data/lab_array_a_10.npy' to '/home/runner/.cache/scikit-image/main'.
```
Unsurprisingly, the expected file doesn't live at https://github.com/scikit-image/scikit-image/raw/main/skimage/color/tests/data/ (yet) since `skimage/color/tests/data/lab_array_a_10.npy` ships with this PR!

I know I didn't run into any issue with Pooch when I added data files (#4939 merged on Sep 2, 2020). Actually, it is the first time (with this PR) we've added files in the data registry since we started running tests with GitHub Actions. Indeed, the time sequence goes like this:

* Oct 31, 2020: merge latest additions to data registry (#5041)
* Nov 26, 2020: start running test suite with GitHub Actions (#5074)

I need to dig deeper to find out how Pooch builds paths, unless someone else at @scikit-image/core knows it right away. Thanks!

**Comment by stefanv** [comment]
@mkcor [This change](https://github.com/scikit-image/scikit-image/pull/5243/files#diff-51849a66eb186097a90aaeda5be0995b0bd3943513593c032bdf1742de21df9fR153) was recently made—I hope that's not the culprit.

**Comment by mkcor** [comment]
> @mkcor [This change](https://github.com/scikit-image/scikit-image/pull/5243/files#diff-51849a66eb186097a90aaeda5be0995b0bd3943513593c032bdf1742de21df9fR153) was recently made—I hope that's not the culprit.

@stefanv I know, initially I thought about that, but apparently it's a different issue: CI tests initially ran 2 weeks ago, before this change was made, and failed in the same way. Thanks though.

File: skimage/data/_registry.py [hunk 3]
```diff
@@ -41,26 +41,54 @@
...
+    "color/tests/data/luv_array_d50_r.npy": "f93f0def9c93f872dd10ce4a91fdb3f06eea61ddb6e72387b7669909827d4f9c",
     "color/tests/data/luv_array_d55_10.npy": "d88d53d2bad230c2331442187712ec52ffdee62bf0f60b200c33411bfed76c60",
     "color/tests/data/luv_array_d55_2.npy": "c761b40475df591ae9c0475d54ef712d067190ca4652efc6308b69080a652061",
+    "color/tests/data/luv_array_d55_r.npy": "05fbd57e3602ee4d5202b9f18f9b5fc05b545891a9b4456d2a88aa798a5a774a",
     "color/tests/data/luv_array_d65_10.npy": "41a5452ffac4d31dd579d9528e725432c60d77b5f505d801898d9401429c89bf",
     "color/tests/data/luv_array_d65_2.npy": "962ce180132c6c11798cbc423b2b204d1d10187670f6eb5dec1058eaad301e0e",
+    "color/tests/data/luv_array_d65_r.npy": "78db8c19af26dd802ce98b039a33855f7c8d6a103a2721d094b1d9c619717449",
     "color/tests/data/luv_array_d75_10.npy": "e1cc70d56eb6789633d4c2a4059b9533f616a7c8592c9bd342403e41d72f45e4",
     "color/tests/data/luv_array_d75_2.npy": "07db3bd59bd89de8e5ff62dad786fe5f4b299133495ba9bea30495b375133a98",
+    "color/tests/data/luv_array_d75_r.npy": "07db3bd59bd89de8e5ff62dad786fe5f4b299133495ba9bea30495b375133a98",
```
Comment (mkcor) [inline skimage/data/_registry.py:88]: hash must be wrong since it's the same as line above

File: skimage/data/_registry.py [hunk 4]
```diff
@@ -41,26 +41,54 @@
...
+    "color/tests/data/luv_array_e_10.npy": "41b1037d81b267305ffe9e8e97e0affa9fa54b18e60413b01b8f11861cb32213",
```
Comment (mkcor) [inline skimage/data/_registry.py:89]: hash must be wrong since it's the same as below

File: skimage/data/_registry.py [hunk 5]
```diff
@@ -41,26 +41,54 @@
...
     "color/tests/data/luv_array_e_2.npy": "41b1037d81b267305ffe9e8e97e0affa9fa54b18e60413b01b8f11861cb32213",
+    "color/tests/data/luv_array_e_r.npy": "41b1037d81b267305ffe9e8e97e0affa9fa54b18e60413b01b8f11861cb32213",
```
Comment (mkcor) [inline skimage/data/_registry.py:91]: hash must be wrong since it's the same as above

File: skimage/data/_registry.py (hunk 4 above)
Comment (mkcor) [reply skimage/data/_registry.py:89]: oh ok, files are the same

File: skimage/data/_registry.py (hunk 5 above)
Comment (mkcor) [reply skimage/data/_registry.py:91]: oh ok, files are the same

File: skimage/data/_registry.py (hunk 3 above)
Comment (mkcor) [reply skimage/data/_registry.py:88]: oh ok, files are the same

**Review by hmaarrfk — APPROVED** [review]
This PR looks solid.

The only thing missing is a release note, but we don't have a good place to put them.

**Comment by hmaarrfk** [comment]
i'm not too worried about the failing tests.

The data is in the right place, and the registry looks great. Thanks for the addition! and the thoroughness!

**Comment by mkcor** [comment]
@hmaarrfk this PR needs a little clean-up before merging (wrt `lab_B_2.npy `, `luv_array_d75_r`, `luv_array_e_r.npy` `luv_array_e_10.npy` as I pointed out, see also my code review comments). But I agree that, *conceptually*, it would be fine to merge since tests would pass on `main`.

I think we can have a clean, linear (sequential) commit history by re-doing the addition of 'new illuminants' from a clean slate, which I have started with #5276. Then, we can document the process as @stefanv suggested, so that each PR (submitted in the proper order) can pass all CI tests.

**Comment by hmaarrfk** [comment]
I think the commit history cleanup is superfluous. We can squash and merge on our end. No need to push that effort onto @BierretA .

The other cleanup is all worthwhile

**Comment by mkcor** [comment]
> No need to push that effort onto @BierretA

@hmaarrfk totally!! Couldn't agree more, that's why I offered to take over at #5276 -- I just forgot to declare @BierretA as a co-author when committing, sorry about that. Will do in subsequent, related commits.

**Comment by alexdesiqueira** [comment]
Assuming that #5308 continues the work started here, and that one was merged, I am closing this one. Thank you @mkcor @BierretA @hmaarrfk for your work!

**Comment by stefanv** [comment]
Thank you @BierretA!

**Pull request closed without merging**